rclc/include/rclc/types.h

```diff
@@ -15,26 +15,46 @@
 #ifndef RCLC__TYPES_H_
 #define RCLC__TYPES_H_
 
-#include "rcl/node.h"
-#include "rcl/publisher.h"
-#include "rcl/subscription.h"
-#include "rcl/types.h"
+#include "rcl/rcl.h"
+
+#include "rclc/type_support.h"
 
 typedef rcl_ret_t rclc_ret_t;
-typedef rcl_node_t rclc_node_t;
-typedef rcl_publisher_t rclc_publisher_t;
+
+typedef struct rclc_node_t rclc_node_t;
+
+typedef struct rclc_publisher_t rclc_publisher_t;
+
+typedef struct rclc_subscription_t rclc_subscription_t;
+
+typedef struct rclc_executor_t rclc_executor_t;
 
 typedef void (* rclc_callback_t)(const void *);
 
-typedef struct rclc_subscription_t
+// impl
+struct rclc_subscription_t
 {
-  struct rclc_subscription_t * rcl_subscription;
+  rcl_subscription_t rcl_subscription;
   rclc_callback_t user_callback;
-} rclc_subscription_t;
 
-typedef struct rclc_executor_t
+  rclc_type_support_t type_support;
+
+  rclc_node_t * node;
+};
+
+struct rclc_publisher_t
 {
-  // TODO(wjwwood): Fill this out
-} rclc_executor_t;
```

Comment:
can you keep the references to the executor ? (as this is a structure we will need to implement in the future)

Reply:
The exectuor struct is still declared [here](https://github.com/ros2/rclc/pull/8/files/b8d58ad2a208a7831407c384d6a983d4d8a6266e#diff-d67cb59d558b3a3f12d6c03e1c71256bR30). (not defined, though)